src/data/all-abilities.ts

```diff
@@ -665,11 +672,12 @@ export function initAbilities() {
     new Ability(Abilities.UNAWARE, 4)
       .attr(IgnoreOpponentStatStagesAbAttr, [Stat.ATK, Stat.DEF, Stat.SPATK, Stat.SPDEF, Stat.ACC, Stat.EVA])
       .ignorable(),
-    new Ability(Abilities.TINTED_LENS, 4).attr(
-      DamageBoostAbAttr,
-      2,
-      (user, target, move) => (target?.getMoveEffectiveness(user!, move) ?? 1) <= 0.5,
-    ),
+    new Ability(Abilities.TINTED_LENS, 4).attr(DamageBoostAbAttr, 2, (user, target, move) => {
+      if (!user || !target || !move) {
+        return false;
+      }
+      return (target.getMoveEffectiveness(user, move) ?? 1) <= 0.5;
```

Comment:
```suggestion
      return target.getMoveEffectiveness(user, move) <= 0.5;
```
This `??` is not be necessary anymore due to the `if (!...)` that was added.